src/ui/public/vislib/lib/axis_config.js

```diff
@@ -86,8 +87,8 @@ export default function AxisConfigFactory() {
       // horizontal axis with ordinal scale should have labels rotated (so we can fit more)
       // unless explicitly overriden by user
       if (this.isHorizontal() && this.isOrdinal()) {
-        this._values.labels.filter = config.labels.filter || false;
-        this._values.labels.rotate = config.labels.rotate || 70;
+        this._values.labels.filter = axisConfig.labels.filter || false;
```

Comment:
Didn't notice this before, but we should probably be reading `axisConfig.labels.filter` from `this._values` right? We don't know if `axisConfig.labels` is even defined, right?